src/main/java/org/zalando/fahrschein/CursorManager.java

```diff
@@ -32,21 +38,25 @@ default void fromNewestAvailableOffsets(String eventName, List<Partition> partit
     }
 
     /**
-     * Initializes offsets to start streaming at the oldes available offset (BEGIN).
+     * Initializes offsets to start streaming at the oldest available offset (BEGIN).
      */
     default void fromOldestAvailableOffset(String eventName, List<Partition> partitions) throws IOException {
         for (Partition partition : partitions) {
-            onSuccess(eventName, new Cursor(partition.getPartition(), partition.getOldestAvailableOffset()));
+            onSuccess(eventName, new Cursor(partition.getPartition(), "BEGIN"));
         }
     }
 
+    /**
+     * Updates offsets in case the currently stored offset is no longer available. Streaming will start at the oldest available offset (BEGIN) to minimize the amount of events skipped.
+     */
     default void updatePartitions(String eventName, List<Partition> partitions) throws IOException {
+
         final Map<String, Cursor> cursorsByPartition = getCursors(eventName).stream().collect(Collectors.toMap(Cursor::getPartition, c -> c));
 
         for (Partition partition : partitions) {
             final Cursor cursor = cursorsByPartition.get(partition.getPartition());
-            if (cursor == null || !partition.isAvailable(cursor.getOffset())) {
-                onSuccess(eventName, new Cursor(partition.getPartition(), partition.getOldestAvailableOffset()));
+            if (cursor == null || (!"BEGIN".equals(cursor.getOffset()) && OFFSET_ORDERING.compare(cursor.getOffset(), partition.getOldestAvailableOffset()) < 0)) {
```

Comment:
> What about extracting ... to a meaningful method?

Then I would have to think about a name and place for the method :)

I prefer to not expose it as a public api, which means it can't be in this interface or the domain model.